python/GafferSceneUI/LightEditor.py

```diff
@@ -352,6 +353,69 @@ def __editSelectedCells( self, pathListing ) :
 
 			self.__popup.popup()
 
+	def __buttonPress( self, pathListing, event ) :
+
+		if event.button != event.Buttons.Right or event.modifiers != event.Modifiers.None_ :
+			return False
+
+		selection = pathListing.getSelection()
+
+		columns = pathListing.getColumns()
+		cellColumn = pathListing.columnAt( event.line.p0 )
+		columnIndex = -1
+		for i in range( 0, len( columns ) ) :
+			if cellColumn == columns[i] :
+				columnIndex = i
+
+		if columnIndex == -1 :
+			return False
+
+		cellPath = pathListing.pathAt( event.line.p0 )
+
+		if str( cellPath ) not in selection[columnIndex].paths() :
```

Comment:
This is copying the selection into a list and then doing a linear search in the list, which has much worse performance than this :

```not selection[columnIndex].match( str( cellPath ) ) & PathMatcher.ExactMatch```

Reply:
Fixed in 6d30a4c7b7d4671509b29ff57937016d63252b30